### ISSUE TITLE
A rule to identify self-closed tags that shouldn't be

### ISSUE THREAD

**Issue by abejfehr** [htmlhint/HTMLHint#950]
**Is your feature request related to a problem? Please describe.**
jQuery recently made a breaking change, detailed [here](https://jquery.com/upgrade-guide/3.5/).

The new behaviour is that sibling tags that are self-closing will now contain each other, for example:

```
<div /><span />
```

will now become:

```
<div><span></span></div>
```

which is pretty unintuitive and requires me to change my Angular templates in many places to no longer have self closing tags.

**Describe the solution you'd like**

What I'd like is for a rule to warn me about self closed `div`, `span`, `my-custom-tag` tags which are negatively impacted when they're self closed.

There's a rule that's _similar_ to what I want, [empty-tag-no-self-closed](https://htmlhint.com/docs/user-guide/rules/empty-tag-not-self-closed) but it only flags elements I don't care about (i.e. I don't care if a `hr`, `meta`, `br` tag, etc. is self closed or not, since they can't have children).

**Describe alternatives you've considered**
I was thinking of making my own rule by forking `empty-tag-no-self-closed`, or maybe there's a way it could be configured to target specific elements.

**Comment by abejfehr** [comment]
I'm trying the "custom rule" approach, but there don't seem to be examples on how the rules in a custom `--rulesdir` should look. I found [this](https://github.com/htmlhint/htmlhint-loader#rulesdir) on a different repo and that's the closest thing, but specifying `--rulesdir ./rules/` doesn't seem to do anything.

I can see that the JS in my rule file is running by putting a `console.log` on line 1, but the `HTMLHint.addRule` method never seems to be executed.

**Comment by abejfehr** [comment]
Actually [tag-pair](https://github.com/htmlhint/HTMLHint/blob/master/dist%2Fcore%2Frules%2Ftag-pair.js) seems to be the rule that I want, but it looks like it doesn't warn for custom elements as per #142

**Comment by abejfehr** [comment]
If anyone's curious, rules in your custom rulesdir must look like this:

```
module.exports = HTMLHint => {
  HTMLHint.addRule({
    id: 'my-rule-id',
    description: 'Description of my rule',
    init(parser, reporter) {
      // rule contents here
    },
  });
};
```

I've created a local rule called `better-tag-pair` that resolves #142 and I'm using that.
